addons/survey/models/survey_survey.py

```diff
@@ -434,6 +495,70 @@ def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
             questions = questions & answer.predefined_question_ids
         return questions, page_or_question_id
 
+    # ------------------------------------------------------------
+    # SESSIONS MANAGEMENT
+    # ------------------------------------------------------------
+
+    def session_next_question(self):
```

Comment:
It was public because it was initially called from RPC but I changed the behavior and did not make it private (oopsy).

I also renamed the method as your naming is more explicit.